### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Kaghenimbale**
- [ ] Create a class Classroom
- [ ] Create the has-many/belongs-to relationship between Classroom and Student. 
- [ ] Create a class Book
- [ ] Create a class Rental
- [ ] Create the many-to-many (also has-many-through) relationship between Person and Book

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
+end
```
Comment (ABDULALI3468) [inline book.rb]: **1-** Please consider updating the Book class to have Getters and Setters both for @rentals to meet the project requirements. 

```
 attr_accessor :title, :author, :rentals
```

**2-** Moreover, for the add_rental method the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id, :rental
```
Comment (ABDULALI3468) [inline person.rb]: **1-** Please consider updating the Person class to have Getters and Setters both for @rentals to meet the project requirements. ✅ 

```
`attr_accessor :name, :age, :rentals`
```

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    true
+  end
+
+  def correct_name
+    @name
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (ABDULALI3468) [inline person.rb]: **1-** Moreover, for the add_rental method The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association

```
 def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by ABDULALI3468 — CHANGES_REQUESTED** [review]
# Changes Required 👍 

Hi @Kaghenimbale , 



Indeed you did an exceptional but still, there is room for changes to complete this project. 💯


## Highlights 
☑️Implemented associations very well.
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.


 ## Required Changes ♻️

> **_Check the comments below to have a look at the required changes._**

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification.

> * This is My **GitHub**: GitHub followers
> * This is My **Linkedin**: [Linkedin Badge](https://www.linkedin.com/in/abdul-ali-5400bb216/)&nbsp;
> * This is My **Twitter**: [Twitter Badge](https://twitter.com/mrabdul_ali)&nbsp;

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by osamaashraf6 — APPROVED** [review]
Hi @Kaghenimbale 😊, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
MinionsGIF (2)

## Highlights🎉🎉

- [x] No linter errors present💯
- [x] The PR has a proper title and a descriptive body 👍

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**